src/Grpc.AspNetCore.Server/Internal/DefaultGrpcInterceptorActivator.cs

```diff
@@ -0,0 +1,66 @@
+﻿#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System;
+using System.Diagnostics;
+using Grpc.Core.Interceptors;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Grpc.AspNetCore.Server.Internal
+{
+    internal class DefaultGrpcInterceptorActivator<TInterceptor> : IGrpcInterceptorActivator<TInterceptor> where TInterceptor : Interceptor
+    {
+        private readonly IServiceProvider _serviceProvider;
+        private bool? _created;
+
+        public DefaultGrpcInterceptorActivator(IServiceProvider serviceProvider)
+        {
+            _serviceProvider = serviceProvider;
+        }
+
+        public TInterceptor Create(params object[] args)
+        {
+            Debug.Assert(!_created.HasValue, "Grpc interceptor activator must not be reused.");
+
+            _created = false;
+            var interceptor = _serviceProvider.GetService<TInterceptor>();
+            if (interceptor == null)
+            {
+                interceptor = ActivatorUtilities.CreateInstance<TInterceptor>(_serviceProvider, args); // TODO: Is there a way to factory-ize this?
```

Comment:
Can you use ActivatorUtilities.CreateFactory? Convert the arg values to type with GetType.

Reply:
There are some interesting implications here. For example, it's possible in the c-core pipeline to add two instances of the same interceptor that were constructed with different args to the pipeline. If we were to support that, I don't think we can use the same trick as DefaultGrpcServiceActivator to create a static factory, in which case compiling a lambda every call may be more inefficient.

Reply:
It is not perfectly efficient but I doubt it will be a problem. How many interceptors is even a very large app going to register? 20? 50?

I think it is an acceptable way of doing it for now, and we can change the delegate caching later if it does become problematic.